core/src/main/java/io/questdb/cairo/pool/ReaderPool.java

```diff
@@ -369,10 +371,7 @@ public void close() {
                 if (pool != null && entry != null && pool.returnToPool(this)) {
```

Comment:
I think `pool` has to be captured on stack. If I insert pause after `null` check but before call to `returnToPool` I get NPE.

```
                if (pool != null && entry != null) {
                    Os.pause();
                    if (pool.returnToPool(this)) {
                        return;
                    }
                }
                super.close();

```

Reply:
Good spot, thank you 🙏 Fixed here: https://github.com/questdb/questdb/pull/1821/commits/ab43855402ecdfd42d14afd7e3f4a12e6e749a2c

Before the fix 2min run would produce NPE. After fix, 10min run - no issues